ddtrace/internal/datadog/profiling/stack_v2/src/stack_renderer.cpp

```diff
@@ -29,18 +29,63 @@ StackRenderer::render_thread_begin(PyThreadState* tstate,
         return;
     }
 
+    ddup_push_threadinfo(sample, static_cast<int64_t>(thread_id), static_cast<int64_t>(native_id), name);
+    ddup_push_walltime(sample, 1000 * wall_time_us, 1);
+
     // Get the current time in ns in a way compatible with python's time.monotonic_ns(), which is backed by
     // clock_gettime(CLOCK_MONOTONIC) on linux and mach_absolute_time() on macOS.
     // This is not the same as std::chrono::steady_clock, which is backed by clock_gettime(CLOCK_MONOTONIC_RAW)
     // (although this is underspecified in the standard)
+    int64_t now_ns = 0;
     timespec ts;
     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
-        auto now_ns = static_cast<int64_t>(ts.tv_sec) * 1'000'000'000LL + static_cast<int64_t>(ts.tv_nsec);
+        now_ns = static_cast<int64_t>(ts.tv_sec) * 1'000'000'000LL + static_cast<int64_t>(ts.tv_nsec);
         ddup_push_monotonic_ns(sample, now_ns);
     }
 
+    // Save the thread information in case we observe a task on the thread
+    thread_state.id = thread_id;
+    thread_state.native_id = native_id;
+    thread_state.name = std::string(name);
+    thread_state.now_time_ns = now_ns;
+    thread_state.wall_time_ns = 1000LL * wall_time_us;
+    thread_state.cpu_time_ns = 0; // Walltime samples are guaranteed, but CPU times are not. Initialize to 0
+                                  // since we don't know if we'll get a CPU time here.
+
+    // Finalize the thread information we have
     ddup_push_threadinfo(sample, static_cast<int64_t>(thread_id), static_cast<int64_t>(native_id), name);
```

Comment:
We're calling `ddup_push_threadinfo` twice in this function. Looking at the `push_label` definition [here](https://github.com/DataDog/dd-trace-py/blob/6e937a8521752179410e1e5e25a9b1e744815491/ddtrace/internal/datadog/profiling/dd_wrapper/src/sample.cpp#L89), there will be two labels with same keys. Do you know how this is handled in the backend and rest of the stack?

Reply:
Yup, that was a mistake.

As for the question, you're correct that it'll push the duplicates all the way down to libdatadog.  I'm not 100% sure what libdatadog will do in that case.  Taking the first, taking the last, and pushing duplicate entries all feel like valid options at some level.  If duplicate tags are pushed all the way to the backend, I don't know what it will do!